some code I added to make  OpenApi reader work on my files

bocasti opened the pull request.

Commit:
some code I added to make  OpenApi reader work on my files

noha commented:
I don't get what this should fix. The primitive schema should not be visited because it rather considered to be abstract. As for the OALink this covers existing bugs. Can you provide me with the file you want to read instead?

Commit:
added example with minItems:

Commit:
implemented tests using #minItems

Commit:
added test illustrating #acceptOpenApi: for string format

Commit:
added test for responses with links

Commit:
renamed test

bocasti commented:
So I added a test for each case:

- testParsingSchemaMinItems => for the #minItems: method
- testParsingResponseWithLinks => for the OALink mapping
- testParsingOpenAPIWithEnum => for the #acceptOpenApi: method

noha commented:
I don't get why you make extension methods here for JSONSchema. Look at https://github.com/zweidenker/JSONSchema/pull/40 that adds minItems and maxItems. What else needs to be done?

noha commented:
Can remove the minItem and json schema stuff please?

Commit:
removed useless test for #minItems:

Commit:
removed hack attempts in order to let test fail and find appropriate fixes

bocasti commented:
Ok, i just
- removed one useless test related to this #miItems: stuff
- removed my "hacks" in OALink and PrimitiveSchema

There should now be 2 test failing, which i think highlights 2 issues
